src/OrchardCore.Modules/OrchardCore.Users/Controllers/AccountController.cs

```diff
@@ -179,5 +205,136 @@ private IActionResult RedirectToLocal(string returnUrl)
                 return Redirect("~/");
             }
         }
+
+        [HttpGet]
+        [AllowAnonymous]
+        public async Task<IActionResult> ForgotPassword()
+        {
+            if (!(await _siteService.GetSiteSettingsAsync()).As<RegistrationSettings>().EnableLostPassword)
+            {
+                return NotFound();
+            }
+
+            return View();
+        }
+
+        [HttpPost]
+        [AllowAnonymous]
+        public async Task<IActionResult> ForgotPassword(ForgotPasswordViewModel model)
+        {
+            if (ModelState.IsValid)
+            {
+                var user = (User)await _userService.GetForgotPasswordUserAsync(model.UserIdentifier);
+                if (user != null)
+                {
+                    user.ResetToken = Convert.ToBase64String(Encoding.UTF8.GetBytes(user.ResetToken));
+                    // send email with callback link
+                    await SendResetTokenAsync(user);
+                }
+            }
+
+            // returns to confirmation page anyway: we don't want to let scrapers know if a username or an email exist
+            return RedirectToLocal(Url.Action("ForgotPasswordConfirmation"));
+        }
+        
+        [HttpGet]
+        [AllowAnonymous]
+        public IActionResult ForgotPasswordConfirmation()
+        {
+            return View();
+        }
+        
+        [HttpGet]
+        [AllowAnonymous]
+        public async Task<IActionResult> ResetPassword(string code = null)
+        {
+            if (!(await _siteService.GetSiteSettingsAsync()).As<RegistrationSettings>().EnableLostPassword)
+            {
+                return NotFound();
+            }
+            if (code == null)
+            {
+                //"A code must be supplied for password reset.";
+            }
+            return View(new ResetPasswordViewModel { ResetToken = code });
+        }
+
+        [HttpPost]
+        [AllowAnonymous]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> ResetPassword(ResetPasswordViewModel model)
+        {
+            if (ModelState.IsValid)
```

Comment:
And then you can reset password without checking
`_siteService.GetSiteSettingsAsync()).As<RegistrationSettings>().EnableLostPassword` via for instance Postman.

Reply:
but hmm... AntiForgeryToken can protect it...